frontend/components/Cell.js

```diff
@@ -48,23 +47,20 @@ export const Cell = ({
     cell_dependencies,
     cell_input_local,
     notebook_id,
-    on_update_doc_query,
-    on_change,
-    on_focus_neighbor,
     selected,
-    selected_cells,
     force_hide_input,
     focus_after_creation,
     is_process_ready,
     disable_input,
     nbpkg,
+    variables_in_all_notebook,
 }) => {
     let pluto_actions = useContext(PlutoContext)
+    const on_update_doc_query = pluto_actions.set_doc_query
+    const on_focus_neighbor = pluto_actions.focus_on_neighbor
+    const on_change = useCallback((val) => pluto_actions.set_local_cell(cell_id, val), [cell_id, pluto_actions])
     const notebook = pluto_actions.get_notebook()
-    let variables_in_all_notebook = Object.fromEntries(
```

Comment:
This creates a new object in each render! We pushed that into the notebook component as it only needs to be calculated once!